src/main/java/org/symphonyoss/symphony/messageml/elements/Radio.java

```diff
@@ -0,0 +1,190 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.symphonyoss.symphony.messageml.elements;
+
+import org.symphonyoss.symphony.messageml.MessageMLParser;
+import org.symphonyoss.symphony.messageml.exceptions.InvalidInputException;
+import org.symphonyoss.symphony.messageml.exceptions.ProcessingException;
+import org.symphonyoss.symphony.messageml.markdown.nodes.form.RadioNode;
+import org.symphonyoss.symphony.messageml.util.XmlPrintStream;
+import org.w3c.dom.NamedNodeMap;
+import org.w3c.dom.NodeList;
+
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * Class representing Radio Buttons - Symphony Elements.
+ *
+ * @author Pedro Sanchez
+ * @since 06/13/19
+ */
+public class Radio extends FormElement {
+
+  public static final String MESSAGEML_TAG = "radio";
+
+  private static final String PRESENTATIONML_INPUT_TYPE = "radio";
+  private static final String PRESENTATIONML_DIV_CLASS = "radio-group";
+  private static final String PRESENTATIONML_LABEL_TAG = "label";
+  private static final int PRESENTATIONML_DIV_NUMBER_OF_CHILDREN = 2;
+  private static final String PRESENTATIONML_DIV_TAG = "div";
+  private static final String PRESENTATIONML_CLASS_ATTR = "class";
+  private static final String PRESENTATIONML_DEFAULT_RADIO_VALUE = "on";
+
+  private static final String NAME_ATTR = "name";
+  private static final String VALUE_ATTR = "value";
+  private static final String CHECKED_ATTR = "checked";
+
+  public Radio(Element parent, FormatEnum messageFormat) {
+    super(parent, MESSAGEML_TAG, messageFormat);
+  }
+
+  @Override
+  public void buildAll(MessageMLParser context, org.w3c.dom.Element element) throws InvalidInputException,
+      ProcessingException {
+    switch (getFormat()) {
+      case MESSAGEML:
+        super.buildAll(context, element);
+        break;
+      case PRESENTATIONML:
+        buildElementFromGroupDiv(context, element);
+        this.validate();
+        break;
+      default:
+        throw new InvalidInputException(String.format("Invalid message format for \"%s\" element", MESSAGEML_TAG));
+    }
+  }
+
+  @Override
+  public org.commonmark.node.Node asMarkdown() {
+    return new RadioNode();
+  }
+
+  @Override
+  public void asPresentationML(XmlPrintStream out) {
+    out.openElement(PRESENTATIONML_DIV_TAG, PRESENTATIONML_CLASS_ATTR, PRESENTATIONML_DIV_CLASS);
+
+    Map<String, String> presentationAttrs = buildRadioInputAttributes();
+    out.printElement(INPUT_TAG, presentationAttrs);
+
+    out.openElement(PRESENTATIONML_LABEL_TAG);
+    for (Element child : getChildren()) {
+      child.asPresentationML(out);
+    }
+    out.closeElement(); // Closing label
+
+    out.closeElement(); // Closing div
+  }
+
+  @Override
+  public void validate() throws InvalidInputException {
+    super.validate();
+    assertRequired(NAME_ATTR);
```

Comment:
```suggestion
    assertRequiredNotBlank(NAME_ATTR);
```

Reply:
It could also be `assertAttributeNotBlank`.

Reply:
changed method to assertAttributeNotBlank